ai2thor/controller.py

```diff
@@ -664,34 +664,13 @@ def step(self, action=None, **action_args):
         if 'AI2THOR_VISIBILITY_DISTANCE' in os.environ:
             action['visibilityDistance'] = float(os.environ['AI2THOR_VISIBILITY_DISTANCE'])
 
-        should_fail = False
         self.last_action = action
 
-        if ('objectId' in action and (action['action'] == 'OpenObject' or action['action'] == 'CloseObject')):
```

Comment:
This removes a "short" circuiting call that effectively fails if an opened object tries to be reopened. It was previously executed after _every single step_, regardless of the action, and had a marginal effect, at best.

Reply:
This short circuit also results in a failure if the `objectId` is not in the set of `instance_dections2D` (if `instance_detections2D` is available). Any idea why that was necessary (@winthos, @ekolve)? Are there really cases where an object is visible but the instance isn't visible? In either case, the behavior should probably not change depending on if `instance_detections2D` being available so I support this removal.

Reply:
(Also, a _terrible_ pain to debug, when everything works perfectly in Unity, but not in Python).

Reply:
I can't think of a case where an object is visible but the instance would not be visible. I think this should be fine to remove. I can also confirm it was a terrible pain to debug.

Reply:
DanielG added this - i believe the intent of all this was just to speed up Open/CloseObject.  I think the instance_detections2d check was basically a crude visibility check (i.e. fail if trying to open the object if its behind you).  All of this should be removed and our rules should be enforced in a single place.